File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/map/ScreenLocationData.java
package com.valentingrigorean.arcgis_maps_flutter.map;

import android.graphics.Point;

import com.esri.arcgisruntime.geometry.SpatialReference;


public class ScreenLocationData {
    private final Point point;
    private final SpatialReference spatialReference;


    public ScreenLocationData(Point point, SpatialReference spatialReference) {
        this.point = point;
        this.spatialReference = spatialReference;
    }

    public Point getPoint() {
        return point;
    }

    public SpatialReference getSpatialReference() {
        return spatialReference;
    }
}

File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/map/ScreenLocationData.kt
package com.valentingrigorean.arcgis_maps_flutter.map

import com.arcgismaps.geometry.SpatialReference
import com.arcgismaps.mapping.view.ScreenCoordinate
import com.valentingrigorean.arcgis_maps_flutter.convert.geometry.toSpatialReferenceOrNull

class ScreenLocationData(val point: ScreenCoordinate, val spatialReference: SpatialReference?) {
    companion object {
        fun fromJson(data: Map<*, *>): ScreenLocationData {
            val points = data["position"] as List<Double>
            val spatialReference = data["spatialReference"]?.toSpatialReferenceOrNull()
            return ScreenLocationData(ScreenCoordinate(points[0], points[1]), spatialReference)
        }
    }
}